### DISCUSSION TITLE
Feature Request: S3 Glob fetch files in parallel

### DISCUSSION THREAD

**Discussion by cmackenzie1** [duckdb/duckdb#6070] [Ideas]
## Why do you want this feature?

Currently, I have hundreds of smaller files stored as Parquet in S3. When using the glob style query it appears to download files sequentially instead of in parallel resulting in queries taking several minutes to complete.  

```sql 
SELECT * FROM read_parquet('s3://bucket/*.parquet)
```

**Comment by cmackenzie1** [comment]
I see there are values for tuning the `s3_uploader_*` but couldn't find any for tuning downloads

**Comment by samansmink** [comment]
Hi @cmackenzie1, this should be fixed, I improved this in https://github.com/duckdb/duckdb/pull/5405 which will be in the v0.7.0 release!

**Comment by ricksmit3000** [comment]
@samansmink does this also aply to a list of files (e.g. `SELECT * FROM read_csv_auto(['flights1.csv', 'flights2.csv'])`) or are those sequential by design? I'm using:
```
Database: v0.8.2-dev1781
Package:  @duckdb/duckdb-wasm@1.27.1-dev59.0
```
and my request waterfall looks like:
<img width="1010" alt="Screenshot 2023-08-28 at 13 45 54" src="https://github.com/duckdb/duckdb/assets/7207783/daaa6c5e-a920-4126-87ed-3e89b58cc267">

**Comment by samansmink** [comment]
Hey @rickiesmooth thanks for the report! duckdb-wasm is a different beast here. It has a different filesystem implementation. I'm not sure whats the current status of duckdb-wasm here, but your screenshot looks similar to https://github.com/duckdb/duckdb-wasm/issues/1087. Would you mind posting this there instead?

**Comment by ricksmit3000** [comment]
ah missed that one! I'll post it there, thanks for checking 🙌
